xinference/deploy/supervisor.py

```diff
@@ -30,10 +32,21 @@ async def start_supervisor_components(address: str, host: str, port: int):
     await xo.create_actor(SupervisorActor, address=address, uid=SupervisorActor.uid())
     gradio_block = GradioApp(address).build()
     # create a socket for RESTful API
-    sockets = []
-    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    sock.bind((host, port))
-    sockets.append(sock)
+    try:
+        sockets = []
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.bind((host, port))
+        sockets.append(sock)
+    except OSError:
+        if port is XINFERENCE_DEFAULT_ENDPOINT_PORT:
```

Comment:
```suggestion
        if port == XINFERENCE_DEFAULT_ENDPOINT_PORT:
```

Reply:
Here I want to differentiate the case between user specify the default port and user does not specify the port, so I comapre their reference.